bin/sirius-script-procedures-machine-shutdown.py

```diff
@@ -0,0 +1,1121 @@
+        # case of any False return.
+        return True
+
+    def s06_sofb_fofb_turnoff(self):
+        """Turn off orbit feedback loops."""
+        self.log('Step 06: Turning off FOFB and SOFB...')
+
+        self.log('Turning off FOFB loop...')
+        is_ok = self._devrefs['fofb'].cmd_turn_off_loop_state(timeout=12)
+        if not is_ok:
+            self.log('ERR:Could not turn off FOFB loop.')
+            return False  # TODO: do we want to abort procedure in this case?
+
+        if not self.continue_execution():
+            return False
+
+        self.log('...done. Turning off SOFB loop...')
+        is_ok = self._devrefs['sofb'].cmd_turn_off_autocorr(timeout=5)
+        if not is_ok:
+            self.log('ERR:Could not turn off SOFB loop.')
+            return False  # TODO: do we want to abort procedure in this case?
+
+        if not self.continue_execution():
+            return False
+
+        self.log('...done. Disabling SOFB Synchronization...')
+        is_ok = self._devrefs['sofb'].cmd_turn_off_synckick(timeout=5)
+        if not is_ok:
+            self.log('WARN:Could not disable SOFB Synchronization.')
+            self.log('WARN:Continuing anyway...')
+            # NOTE: we will not return False here because the correctors
+            # mode will be controlled also in the PS turn of procedure.
+
+        self.log('...done.')
+        return True
+
+    def s07_bbb_turnoff(self):
+        """Turn off BbB loops."""
+        self.log('Step 07: Turning off BbB...')
+
+        self.log('Disabling BbB loops...')
+
+        devices = [
+            self._devrefs[d] for d in ['bbbhfb', 'bbbvfb', 'bbblfb']]
+
+        for dev in devices:
+            dev.loop_state = 0
+
+        if not self._wait_devices_propty(devices, 'FBCTRL', 0):
+            self.log('WARN:Could not disable BbB loops.')
+            self.log('WARN:Continuing anyway...')
+        else:
+            self.log('...done.')
+        return True
+
+    def s08_sirf_turnoff(self):
+        """Turn off SI RF."""
+        self.log('Step 08: Turning off SI RF...')
+
+        llrf = self._devrefs['sillrf']
+        cavmon = self._devrefs['sicavmon']
+        preamp = self._devrefs['sillrfpreamp']
+        dcamp1 = self._devrefs['sirfdcamp1']
+        dcamp2 = self._devrefs['sirfdcamp2']
+        acamp1 = self._devrefs['sirfacamp1']
+        acamp2 = self._devrefs['sirfacamp2']
+
+        self.log('Changing voltage increase rate to 2mV/s...')
+        incrate = llrf.VoltIncRates.vel_2p0
+        if not llrf.set_voltage_incrate(incrate, timeout=5):
+            self.log('ERR:Could not set voltage increase rate to 2mV/s.')
+            return False
+        _time.sleep(1.0)  # is this necessary?
+
+        if not self.continue_execution():
+            return False
+
+        self.log('...done. Changing loop reference to 60mV...')
+        if not llrf.set_voltage(60, timeout=180, wait_mon=False):
```

Comment:
> Wouldn't be better to define a class constant here?

done!

Reply:
> I think we could implement a this turnoff (and also a turn on) procedure in that device.

maybe we could migrate to there, but it could be a future enhancement (I won't be able to work on this before october)